internal/controllers/gateway/gateway_utils.go

```diff
@@ -670,3 +666,71 @@ func isTLSSecretValid(secret *corev1.Secret) bool {
 	}
 	return true
 }
+
+func routeAcceptedByGateways(routeNamespace string, parentStatuses []RouteParentStatus) []k8stypes.NamespacedName {
+	gateways := []k8stypes.NamespacedName{}
+	for _, routeParentStatus := range parentStatuses {
+		gatewayNamespace := routeNamespace
+		parentRef := routeParentStatus.ParentRef
+		if (parentRef.Group != nil && *parentRef.Group != gatewayV1beta1Group) ||
+			(parentRef.Kind != nil && *parentRef.Kind != "Gateway") {
+			continue
+		}
+		if parentRef.Namespace != nil {
+			gatewayNamespace = string(*parentRef.Namespace)
+		}
+		if lo.ContainsBy(routeParentStatus.Conditions, func(condition metav1.Condition) bool {
+			return condition.Type == string(gatewayv1beta1.RouteConditionAccepted) &&
+				condition.Status == metav1.ConditionTrue
+		}) {
+			gateways = append(gateways,
+				k8stypes.NamespacedName{
+					Namespace: gatewayNamespace,
+					Name:      string(parentRef.Name),
+				})
+		}
+	}
+	return gateways
+}
+
+// Get all the HTTPRoutes and update the listeners' attachedRoutes field.
+// At this point we take into account HTTPRoutes only, as they are the
+// only routes in beta.
```

Comment:
This is a loose proposal for rewording. Feel free to suggest something else.

One thing that's missing for me is why do we need to provide both the listener and the gateway. That might be beneficial to document for the reader.

```suggestion
// getAttachedRoutesForListener returns the number of all the routes that are attached
// to the provided Gateway.
//
// NOTE: At this point we take into account HTTPRoutes only, as they are the
// only routes in beta.
```